doc/src/concepts/proxies.md

```diff
@@ -0,0 +1,29 @@
+# Proxies
+
+`rustup` provides a number of wrappers for common Rust tools.
+These are called _proxies_ and represent commands which are
+provided by the various [components][].
+
+The list of proxies is currently static in `rustup` and is as follows:
+
+[components]: concepts/components.html
+
+- `rustc`: `rustc` is the compiler for the Rust programming language, provided by the project itself and come the from the `rustc` component.
+
+- `rustdoc`: `rustdoc` is a tool distributed in the `rustc` component which helps you to generate documentation for Rust projects.
+
+- `cargo`: cargo is the rust package manager which downloads your Rust package’s dependencies, compiles your packages, makes distributable packages, and uploads them to crates.io and the Rust community’s package registry and it comes from the `cargo` component.
+
+- `rust-lldb`: rust-lldb is a debugger component of the LLVM project. It is built as a set of reusable components that extensively use existing libraries from LLVM, such as the Clang expression parser and LLVM disassembler and it comes from the `rustc` component.
+
+- `rust-gdb`: rust-gdb helps you to pretty print the program with GDB python. You can see the detail in [here](https://github.com/rust-lang/rust/blob/master/src/etc/rust-gdb).It comes from the `rustc` component.
```

Comment:
Are these descriptions lifted from somewhere else?  They feel a little disjointed and confusing.  `rust-lldb` is as much a wrapper as `rust-gdb` and they're both there simply to make it easier to debug Rust code using the `lldb` and `gdb` tools respectively.  I'd be tempted to wrap both of these up with a single bullet point of the form:

```markdown
- `rust-lldb` and `rust-gdb` are simple wrappers around the `lldb` and `gdb` debuggers respectively.  The wrappers enable some pretty-printing of Rust values and add some convenience features to the debuggers by means of their scripting interfaces.
````

Reply:
Thank you for your review and help. Yeah, I read the documents of LLDB & GDB and source code and tried to explain it as much as possible but that's too disjoiinted and unhelpful.